sstables_loader.cc

```diff
@@ -343,55 +343,60 @@ struct per_tablet_stream_progress : public stream_progress {
     }
 };
 
-future<> tablet_sstable_streamer::stream(shared_ptr<stream_progress> progress) {
-    if (progress) {
-        progress->start(_tablet_map.tablet_count());
+future<std::vector<tablet_sstable_collection>> get_sstables_for_tablets(const std::vector<sstables::shared_sstable>& sstables,
+                                                                                                 std::vector<tablet_sstable_collection>&& tablets_sstables) {
+    if (sstables.empty() || tablets_sstables.empty()) {
+        co_return std::move(tablets_sstables);
     }
 
     // sstables are sorted by first key in reverse order.
```

Comment:
please improve the comment by saying they arrive in reverse order, but let's operate on normal order to avoid unnecessary brain strokes.
Also, I would name the function parameter `reversed_sstables` just to reflect the fact that they arrived reversed, then apply the reverse view and name that simple sstables; this, combined with a more verbose comment should make the situation clearer.